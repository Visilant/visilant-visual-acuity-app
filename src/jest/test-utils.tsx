import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { fireEvent, render, RenderOptions, screen } from '@testing-library/react-native'
import { WrapperProps } from './types'
import { ThemeProvider } from 'styled-components/native'
import { appTheme } from '@components/theme/theme'
import { LocalizationProvider } from '@localizations/Provider'
import { store } from '@store'

const Providers: React.FunctionComponent<WrapperProps> = ({ children }) => {
  return (
    <ReduxProvider store={store}>
      <ThemeProvider theme={appTheme}>
        <LocalizationProvider>{children}</LocalizationProvider>
      </ThemeProvider>
    </ReduxProvider>
  )
}

const customRender = <T,>(
  ui: React.ReactElement<unknown, string | React.JSXElementConstructor<T>>,
  options?: RenderOptions
) => {
  return render(ui, {
    wrapper: Providers,
    ...options
  })
}

export { fireEvent, customRender as render, screen }
