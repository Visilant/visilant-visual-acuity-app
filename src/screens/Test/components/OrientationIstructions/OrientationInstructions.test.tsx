import { fireEvent, render, screen } from '@jest/test-utils'
import { OrientationInstructions } from '@screens/Test/components/OrientationIstructions/OrientationInstructions'
const mockNavigate = jest.fn()
jest.mock('@screens/stack', () => ({ useStackNavigation: () => ({ navigate: mockNavigate }) }))

describe('Orientation Instructions Screen', () => {
  test('matches the snapshot', async () => {
    const orientationScreen = render(<OrientationInstructions />)

    expect(orientationScreen).toMatchSnapshot()
  })

  test('the Home button navigates away', () => {
    render(<OrientationInstructions />)
    const btn = screen.getByTestId('orientation-instructions.home')
    fireEvent(btn, 'press')

    expect(mockNavigate).toBeCalledTimes(1)
  })

  test('renders instructions', async () => {
    render(<OrientationInstructions />)
    screen.getByText('Rotate your phone to start the test!')
  })
})
