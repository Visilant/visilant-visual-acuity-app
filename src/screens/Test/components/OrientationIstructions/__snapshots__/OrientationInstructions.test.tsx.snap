// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orientation Instructions Screen matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#EDF1F4",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 1013.5501355013549,
        "marginLeft": -50,
        "marginTop": -100,
        "width": 850,
      }
    }
  />
  <View
    style={
      {
        "height": "100%",
        "justifyContent": "space-between",
        "left": 0,
        "paddingBottom": 60,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 60,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-end",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 25,
            "marginRight": 0,
            "width": 25,
          }
        }
        testID="orientation-instructions.home"
      >
        <svg
          fill="#23255A"
        />
      </View>
    </View>
    <Text
      style={
        [
          {
            "color": "#23255A",
            "fontFamily": "Caprasimo",
            "fontSize": 36,
            "lineHeight": 38,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          "
  ",
          [Function],
          "
  font-family: Caprasimo;
  font-size: 36px;
  line-height: 38px;
",
        ]
      }
    >
      Rotate your phone to start the test!
    </Text>
  </View>
</View>
`;
